packages/contracts-bedrock/scripts/deploy/DeployMIPS.s.sol

```diff
@@ -0,0 +1,230 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+// Forge
+import { Script } from "forge-std/Script.sol";
+
+// Scripts
+import { BaseDeployIO } from "scripts/deploy/BaseDeployIO.sol";
+import { DeployUtils } from "scripts/libraries/DeployUtils.sol";
+
+// Libraries
+import { LibString } from "@solady/utils/LibString.sol";
+
+// Interfaces
+import { IBigStepper } from "src/dispute/interfaces/IBigStepper.sol";
+import { IPreimageOracle } from "src/cannon/interfaces/IPreimageOracle.sol";
+import { IMIPS } from "src/cannon/interfaces/IMIPS.sol";
+
+/// @title DeployMIPSInput
+contract DeployMIPSInput is BaseDeployIO {
+    // Common inputs.
+    string internal _release;
+    string internal _standardVersionsToml;
+
+    // All inputs required to deploy PreimageOracle.
+    uint256 internal _minProposalSizeBytes;
+    uint256 internal _challengePeriodSeconds;
+
+    // Specify which MIPS version to use.
+    uint256 internal _mipsVersion;
+
+    function set(bytes4 _sel, uint256 _value) public {
+        if (_sel == this.mipsVersion.selector) {
+            require(_value == 1 || _value == 2, "DeployMIPS: unknown mips version");
+            _mipsVersion = _value;
+        } else if (_sel == this.minProposalSizeBytes.selector) {
+            require(_value != 0, "DeployMIPS: minProposalSizeBytes cannot be zero");
+            _minProposalSizeBytes = _value;
+        } else if (_sel == this.challengePeriodSeconds.selector) {
+            require(_value != 0, "DeployMIPS: challengePeriodSeconds cannot be zero");
+            _challengePeriodSeconds = _value;
+        } else {
+            revert("DeployMIPS: unknown selector");
+        }
+    }
+
+    function set(bytes4 _sel, string memory _value) public {
+        if (_sel == this.release.selector) {
+            require(!LibString.eq(_value, ""), "DeployMIPS: release cannot be empty");
+            _release = _value;
+        } else if (_sel == this.standardVersionsToml.selector) {
+            require(!LibString.eq(_value, ""), "DeployMIPS: standardVersionsToml cannot be empty");
+            _standardVersionsToml = _value;
+        } else {
+            revert("DeployMIPS: unknown selector");
+        }
+    }
+
+    function release() public view returns (string memory) {
+        require(!LibString.eq(_release, ""), "DeployMIPS: release not set");
+        return _release;
+    }
+
+    function standardVersionsToml() public view returns (string memory) {
+        require(!LibString.eq(_standardVersionsToml, ""), "DeployMIPS: standardVersionsToml not set");
+        return _standardVersionsToml;
+    }
+
+    function mipsVersion() public view returns (uint256) {
+        require(_mipsVersion != 0, "DeployMIPS: mipsVersion not set");
+        require(_mipsVersion == 1 || _mipsVersion == 2, "DeployMIPS: unknown mips version");
```

Comment:
None of the requires in these methods ever actually fail.  You can set `--mips-version=5` and the script still executes perfectly fine.

Reply:
The exact same `require` statements fail if put immediately after this `mipsVersion()` call but the versions in these accessors don't have any effect.

Reply:
Apparently this is a known limitation.